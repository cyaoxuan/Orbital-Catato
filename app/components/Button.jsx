import { View } from "react-native";
import { Button } from "react-native-paper";

const PillButton = (props) => {
    return (
        <View style={{ flexDirection: "row" }}>
                <Button style = {{ margin: 10, width: props.width }}
                    disabled={props.disabled}
                    mode={props.mode}
                    onPress={props.onPress}>
                        {props.label}
                </Button>
        </View>
    )
}

export { PillButton };