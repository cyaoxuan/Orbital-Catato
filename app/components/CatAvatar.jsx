import { StyleSheet, TouchableOpacity, View } from "react-native";
import { Avatar } from "react-native-paper";
import { BodyText, TitleText } from "./Text";

// Cat Avatar with circular image and name
// props: photoURL (string), size (image size || number), variant (font size || string), name (string)
const CatAvatar = (props) => {
    return (
        <View style={styles.avatarContainer}>
            <Avatar.Image
                testID="avatar-image"
                source={
                    props.photoURL
                        ? { uri: props.photoURL }
                        : require("../../assets/placeholder.png")
                }
                size={props.size || 100}
            />
            <BodyText variant={props.variant} text={props.name || "Name"} />
        </View>
    );
};

// Cat Avatar with circular image and name, touchable
// props: photoURL (string), size (image size || number), variant (font size || string), name (string), onPress (callback)
const TouchableCatAvatar = (props) => {
    return (
        <TouchableOpacity
            testID="cat-avatar"
            style={styles.avatarContainer}
            activeOpacity={0.7}
            onPress={props.onPress}
        >
            <Avatar.Image
                testID="avatar-image"
                source={
                    props.photoURL
                        ? { uri: props.photoURL }
                        : require("../../assets/placeholder.png")
                }
                size={props.size || 100}
            />
            <BodyText variant={props.variant} text={props.name || "Name"} />
        </TouchableOpacity>
    );
};

export { CatAvatar, TouchableCatAvatar };

const styles = StyleSheet.create({
    avatarContainer: {
        alignItems: "center",
    },
});
