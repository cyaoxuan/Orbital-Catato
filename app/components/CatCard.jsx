import { Card } from "react-native-paper";
import { IconTextField } from "./InfoText";

const CatCard = (props) => {
    return (
        <Card style={{ height: 280, width: 280, margin: 4 }} 
            onPress={() => {}}
            mode="elevated">
            <Card.Cover style={{ height: 140, width: 280, resizeMode: "cover"}} 
                source={props.catImage} 
            />
            <Card.Title title={props.catName} titleVariant="headlineMedium"/>
            <Card.Content style={{ paddingBottom: 0, paddingHorizontal: 4 }}>
                <IconTextField 
                    iconName={props.iconName1} 
                    iconSize={24} 
                    variant="titleLarge"
                    field={props.field1}
                    info={props.info1}
                />
                <IconTextField 
                    iconName={props.iconName2} 
                    iconSize={24} 
                    variant="titleLarge" 
                    field={props.field2}
                    info={props.info2}
                />
            </Card.Content>
        </Card>
    );
};

export { CatCard };