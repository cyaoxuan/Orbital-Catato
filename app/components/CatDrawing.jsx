import { Image } from "react-native";

export const PantingCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-1.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const PopCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-2.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const SmilingGingerCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-3.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const LoadingCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-4.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const MaxwellCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-5.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const SaladSmudgeCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-6.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const StandingCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-7.png")}
            style={{ height: size, width: size }}
        />
    );
};
