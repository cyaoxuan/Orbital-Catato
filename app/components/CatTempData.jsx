// Temp data to showcase static pages

const catViewProps = [{
    iconName1: "information-circle",
    field1: "Status: ",
    iconName2: "location",
    field2: "Last Seen: "
}, {
    iconName1: "time",
    field1: "Last Fed: ",
    iconName2: "location",
    field2: "Last Seen: "
}];

const concernCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Engineering"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Skitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "New",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Utown"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Mitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Missing",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Science"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Bitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Arts"
}];

const fedCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[1].iconName1,
    field1: catViewProps[1].field1,
    info1: "19/05 12:01",
    iconName2: catViewProps[1].iconName2,
    field2: catViewProps[1].field2,
    info2: "Engineering"
}];

export { concernCatCards, fedCatCards }