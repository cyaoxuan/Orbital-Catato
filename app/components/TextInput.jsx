import { View, StyleSheet } from "react-native";
import { TextInput, HelperText } from "react-native-paper";
import { useState } from "react";
import Ionicons from '@expo/vector-icons/Ionicons';

const AuthInput = (props) => {
    return (
        <View style={styles.inputContainer}>
            <Ionicons name={props.iconName} size={24} style={ styles.icon } />
            <TextInput style={styles.input}
                    mode="outlined"
                    disabled={props.disabled}
                    label={props.label}
                    placeholder={props.placeholder}

                    autoCapitalize="none"
                    textContentType={props.textContentType}

                    value={props.value}
                    onChangeText={props.onChangeText}
                  />
        </View>
    );
};

const PasswordInput = (props) => {
    const [secureText, setSecureText] = useState(true);

    return (
        <View style={styles.inputContainer}>
            <Ionicons name={props.iconName} size={24} style={styles.icon} />
            <TextInput style={styles.input}
                mode="outlined"
                disabled={props.disabled}
                label={props.label}
                placeholder={props.placeholder}

                autoCorrect={false}
                autoCapitalize="none"

                textContentType={props.textContentType}
                value={props.value}
                onChangeText={props.onChangeText}

                secureTextEntry={secureText}
                right={
                    <TextInput.Icon
                        icon={secureText ? "eye" : "eye-off"}
                        onPress={() =>
                            setSecureText((prev) => !prev)
                        }
                        forceTextInputFocus={false}
                    />
                }
            />    
        </View>
    );
};

// Required Input, need to pass in a 'value' prop for the error message to work correctly
const FormInput = (props) => {
    return (
        <View style={{ width: "90%", height:100}}>
            <TextInput style={styles.formInput}
                disabled={props.disabled}
                multiline={props.multiline}
                label={props.label}
                placeholder={props.placeholder}
            
                value={props.value}
                onChangeText={props.onChangeText}

                error={!props.value}
            />
            <HelperText type="error" padding="none" visible={!props.value}>
                Error: {props.errorText}
            </HelperText>
        </View>
    );
};

export { AuthInput, PasswordInput, FormInput };

const styles = StyleSheet.create({
    inputContainer: {
        flexDirection: "row", 
        alignItems: "center",
        justifyContent: "center"
    },
    
    input: {
        fontSize: 16,
        width: "80%",
    },

    formInput: {
        fontSize: 16,
        width: "100%",
        backgroundColor: "transparent",
        paddingHorizontal: 0,
        alignContent: "center"
    },

    icon: {
        marginHorizontal: 8
    }
  });
