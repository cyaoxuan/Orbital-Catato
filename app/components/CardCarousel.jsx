import { View, FlatList } from "react-native";
import { CatCard } from "../components/CatCard";

const boxWidth = 300;

const Carousel = (props) => {
    return (
        <FlatList style={{ height: 300 }}
                horizontal
                snapToAlignment="center"
                decelerationRate="normal"
                showsHorizontalScrollIndicator={false}
                scrollEventThrottle={10}
                snapToInterval={boxWidth}
                contentOffset={{ x: -120, y: 0 }}
                ItemSeparatorComponent={() => <View style={{width: 20}} />}
                
                data={props.cards}
                renderItem={({item, index}) => {
                    return (
                        <View key={index}> 
                            <CatCard 
                                catImage={item.catImage}
                                catName={item.catName}
                                iconName1={item.iconName1}
                                field1={item.field1}
                                info1={item.info1}
                                iconName2={item.iconName2}
                                field2={item.field2}
                                info2={item.info2}
                            />
                        </View>
                        
                    );
                }}
        />

    );
};

export { Carousel }