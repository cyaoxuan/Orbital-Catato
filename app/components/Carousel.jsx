import { FlatList, View } from "react-native";
import { CatCard } from "./CatCard";
import {
    formatLastFedSimple,
    formatLastSeenSimple,
} from "../utils/formatDetails";

// Function to get info1 field on the card
// @param carouselType: "concern" or "fed"
// @param cat: cat object with concernStatus or lastFedTime
// @return formatted string depending on carouselType or "Unknown" if any fields are missing
function getInfo1(carouselType, cat) {
    if (!cat || !carouselType) {
        return "Unknown";
    }

    if (
        carouselType === "concern" &&
        cat.concernStatus &&
        Object.keys(cat.concernStatus).length !== 0
    ) {
        const concerns = [];
        cat.concernStatus.injured && concerns.push("Injured");
        cat.concernStatus.missing && concerns.push("Missing");
        cat.concernStatus.new && concerns.push("New");

        return concerns.join(", ");
    } else if (carouselType === "unfed" && cat.lastFedTime) {
        return formatLastFedSimple(cat.lastFedTime);
    }
    return "Unknown";
}

// Card Carousel used in Dashboard, implemented using FlatList and snaps to each card
// props: cats (array of cat objects), cardWidth (number), carouselType (string), userRole (object of roles), ...card (props for card)
const CardCarousel = ({
    cats,
    cardWidth,
    carouselType,
    userRole,
    navigation,
    ...card
}) => {
    const spaceBetweenCards = 16;

    return (
        <FlatList
            testID="card-carousel"
            style={{ height: (cardWidth * 6) / 5 }}
            horizontal
            snapToAlignment="center"
            decelerationRate="normal"
            showsHorizontalScrollIndicator={false}
            scrollEventThrottle={4}
            snapToInterval={cardWidth + spaceBetweenCards}
            ItemSeparatorComponent={() => (
                <View style={{ width: spaceBetweenCards }} />
            )}
            data={cats}
            keyExtractor={(item, index) => item.catID}
            renderItem={({ item }) => {
                return (
                    <CatCard
                        name={item.name}
                        photoURL={item.photoURLs ? item.photoURLs[0] : null}
                        cardWidth={cardWidth}
                        profileOnPress={() =>
                            navigation.navigate("catalogue", {
                                screen: "CatProfile",
                                initial: false,
                                params: { catID: item.catID },
                            })
                        }
                        {...card}
                        info1={getInfo1(carouselType, item)}
                        info2={
                            item.locationName && item.locationZone && userRole
                                ? formatLastSeenSimple(
                                      userRole.isCaretaker
                                          ? item.locationName
                                          : item.locationZone,
                                      item.lastSeenTime
                                  )
                                : "Unknown"
                        }
                        showFindLocation={item.lastSeenLocation}
                        locationOnPress={() => {
                            navigation.navigate("Map", {
                                catID: item.catID,
                                location: item.lastSeenLocation,
                            });
                        }}
                        userRole={userRole}
                    />
                );
            }}
        />
    );
};

export { getInfo1, CardCarousel };
