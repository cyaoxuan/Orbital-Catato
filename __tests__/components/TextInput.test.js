import { AuthInput, PasswordInput } from "../../app/components/TextInput";
import { cleanup, fireEvent, render } from "@testing-library/react-native";

afterEach(cleanup);

describe("<AuthInput />", () => {
    it("renders successfully", () => {
        const authInput = render(<AuthInput />);
        expect(authInput).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<AuthInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders icon", () => {
        const { getByTestId } = render(<AuthInput />);
        const icon = getByTestId("icon");
        expect(icon).toBeDefined();
    });

    it("apply value when changing text", () => {
        const { getByTestId } = render(<AuthInput />);
        const authInput = getByTestId("input");
        fireEvent.changeText(authInput, "hello");
        expect(authInput.props.value).toBe("hello");
    });

    it("displays label prop", () => {
        const { queryAllByText } = render(<AuthInput label="Meow!" />);
        expect(queryAllByText("Meow!")).not.toBeNull();
    });
});

describe("<PasswordInput />", () => {
    it("renders successfully", () => {
        const keyTextField = render(<PasswordInput />);
        expect(keyTextField).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<PasswordInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the icon", () => {
        const { getByTestId } = render(<AuthInput />);
        const icon = getByTestId("icon");
        expect(icon).toBeDefined();
    });

    it("apply value when changing text", () => {
        const { getByTestId } = render(<PasswordInput />);
        const passwordInput = getByTestId("input");
        fireEvent.changeText(passwordInput, "hello");
        expect(passwordInput.props.value).toBe("hello");
    });

    it("reveal hidden text when click eye icon", () => {
        const { getByTestId } = render(<PasswordInput />);
        const passwordInput = getByTestId("input");
        const hideIcon = getByTestId("hide-icon");

        fireEvent.press(hideIcon);
        expect(passwordInput.props.secureTextEntry).toBe(false);

        fireEvent.press(hideIcon);
        expect(passwordInput.props.secureTextEntry).toBe(true);
    });

    it("displays label prop", () => {
        const { queryAllByText } = render(<PasswordInput label="Meow!" />);
        expect(queryAllByText("Meow!")).not.toBeNull();
    });
});
