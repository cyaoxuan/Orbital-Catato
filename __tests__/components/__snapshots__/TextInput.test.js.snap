// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthInput /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="icon"
  >
    
  </Text>
  <View
    style={
      {
        "width": "80%",
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
          undefined,
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 56,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "backgroundColor": "rgba(255, 251, 254, 1)",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 8,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": -0.6666666666666667,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 33,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                  {
                    "scaleY": 1,
                  },
                ],
                "width": -16,
                "writingDirection": "ltr",
              }
            }
          >
            Label
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              {
                "color": "rgba(103, 80, 164, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="input-label-active"
          >
            Label
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="input-label-inactive"
          >
            Label
          </Text>
        </View>
        <TextInput
          autoCapitalize="none"
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            [
              {
                "margin": 0,
                "zIndex": 1,
              },
              {
                "height": 48,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "paddingHorizontal": 16,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`<PasswordInput /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="icon"
  >
    
  </Text>
  <View
    style={
      {
        "width": "80%",
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
          undefined,
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 56,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "backgroundColor": "rgba(255, 251, 254, 1)",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 8,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": -0.6666666666666667,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 33,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                  {
                    "scaleY": 1,
                  },
                ],
                "width": -16,
                "writingDirection": "ltr",
              }
            }
          >
            Label
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              {
                "color": "rgba(103, 80, 164, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="input-label-active"
          >
            Label
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="input-label-inactive"
          >
            Label
          </Text>
        </View>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          secureTextEntry={true}
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            [
              {
                "margin": 0,
                "zIndex": 1,
              },
              {
                "height": 48,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "paddingHorizontal": 16,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              {
                "marginRight": 56,
                "paddingRight": 0,
              },
              undefined,
            ]
          }
          testID="input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            {
              "right": 16,
              "top": 19,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "height": 40,
              "margin": 0,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "width": 40,
            }
          }
          testID="hide-icon-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "flex": 1,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="hide-icon-container"
          >
            <View
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityTraits="button"
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              centered={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ],
                ]
              }
              testID="hide-icon"
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontSize": 24,
                    },
                    [
                      {
                        "lineHeight": 24,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ],
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰈈
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
